src/lib/operators/aggregate_sort.cpp

```diff
@@ -256,6 +256,69 @@ void AggregateSort::_set_and_write_aggregate_value(
   }
 }
 
+Segments AggregateSort::_get_segments_of_chunk(const std::shared_ptr<const Table>& input_table, const ChunkID chunk_id) {
+  Segments segments{};
+  const auto column_count = input_table->column_count();
+  segments.reserve(column_count);
+  const auto chunk = input_table->get_chunk(chunk_id);
+  for (auto column_id = ColumnID{0}; column_id < column_count; ++column_id) {
+    segments.emplace_back(chunk->get_segment(column_id));
+  }
+  return segments;
+}
+
+std::shared_ptr<Table> AggregateSort::_sort_within_chunks(const std::shared_ptr<const Table>& input_table,
+                                                          const std::vector<ColumnID>& groupby_column_ids) {
+  auto sorted_table =
+      std::make_shared<Table>(input_table->column_definitions(), TableType::References, std::nullopt, UseMvcc::No);
+
+  const auto chunk_count = input_table->chunk_count();
+  for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
+    auto chunk = std::make_shared<Chunk>(_get_segments_of_chunk(input_table, chunk_id));
+    std::vector<std::shared_ptr<Chunk>> single_chunk_to_sort_as_vector = {chunk};
+    auto single_chunk_table = std::make_shared<const Table>(input_table->column_definitions(), input_table->type(),
+                                                            std::move(single_chunk_to_sort_as_vector), UseMvcc::No);
+    std::shared_ptr<const Table> single_chunk_table_sorted;
+
+    // sort whole table by sorting a single-chunk table
+    auto input_sorted_by = input_table->get_chunk(chunk_id)->sorted_by();
+    auto output_sorted_by = input_sorted_by;
```

Comment:
Why?

Reply:
That whole function looks strange. I overlooked quite a lot here. :(

Reply:
Working on a rewrite.